packages/Guard/src/GuardSupervisor.js

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+export const GuardRegistrationContext = React.createContext();
+export const GuardGetekeeperContext = React.createContext();
+
+export default function GuardSupervisor({ alertMessageDefault, children }) {
+  const guards = React.useRef([]);
+  const lastId = React.useRef(0);
+
+  function hasDirtyConnectors(group) {
+    const checkIsDirty = guard => guard.isDirty;
+
+    if (!group) {
+      return guards.current.some(checkIsDirty);
+    }
+
+    const checkIsInGroup = guard => (Array.isArray(group) ? group.indexOf(guard.group) >= 0 : guard.group === group);
+
+    return guards.current.filter(checkIsInGroup).some(checkIsDirty);
+  }
+
+  React.useEffect(() => {
+    function handleWindowBeforeUnload(event) {
+      if (hasDirtyConnectors()) {
+        const message = alertMessageDefault;
+
+        // eslint-disable-next-line no-param-reassign
+        event.returnValue = message;
```

Comment:
What does this line do?

Reply:
Seems like this function only needs to do:
```js
event.preventDefault();
event.returnValue = "";
```
But maybe leaving the `message` value in there is useful for some browsers.